packages/at_secondary_server/test/sync_unit_test.dart

```diff
@@ -0,0 +1,724 @@
+        expect(syncResponse[0]['operation'], '+');
+        expect(syncResponse[0]['metadata']['version'], '0');
+
+        expect(syncResponse[1]['atKey'], 'firstname.buzz@alice');
+        expect(syncResponse[1]['commitId'], 3);
+        expect(syncResponse[1]['operation'], '+');
+        expect(syncResponse[1]['metadata']['version'], '0');
+      });
+      test('test to verify sync response contains only valid keys', () {
+        /// Preconditions:
+        /// 1. The serverCommitId is at 10
+        ///    Following are the keys from commitId 6 to 10
+        ///    commitId:6 -  public:phone.wavi@bob
+        ///    commitId:7 -  @alice:mobile.wavi@bob
+        ///    commitId:8 -  @alice:invalidkey@.wavi@bob
+        ///    commitId:9 -  public:firstName.wavi@bob
+        ///    commitId:10 - @alice:country.wavi@bob
+        /// 2. Sync request received: sync:from:5:pageLimit:10
+        ///
+        /// Operation:
+        /// Secondary server receives sync request
+        ///
+        /// Assertions:
+        /// 1. The sync response should contains following entries in the order
+        ///    commitId:6 -  public:phone.wavi@bob
+        ///    commitId:7 -  @alice:mobile.wavi@bob
+        ///    commitId:9 -  public:firstName.wavi@bob
+        ///    commitId:10 - @alice:country.wavi@bob
+        /// TODO: Needs Refactoring:
+        /// At the moment, we do not perform validation on the keys when adding to sync response.
+        /// To prevent the malformed keys adding to sync response. Should we add a validation on keys
+        /// prior to adding to sync response?
+      });
+      test('test to verify sync response does not exceed the buffer limit',
+          () async {
+        /// Preconditions:
+        /// 1. Initialize the server keystore with 5 keys
+        /// 2. Override the sync buffer to a smaller value : 250 Bytes (Bytes to add single commit entry)
+        /// 3. Sync command received from client: sync:from:-1:limit:10
+        ///
+        /// Operations:
+        /// Process sync verb
+        ///
+        /// Assertions:
+        /// The sync response should not exceed the sync buffer size
+        var syncProgressiveVerbHandler = SyncProgressiveVerbHandler(
+            secondaryPersistenceStore!.getSecondaryKeyStore()!);
+        // Setting buffer size to 250 Bytes
+        syncProgressiveVerbHandler.syncBuffer = ByteBuffer(capacity: 250);
+        var response = Response();
+        var inBoundSessionId = '_6665436c-29ff-481b-8dc6-129e89199718';
+        var atConnection = InboundConnectionImpl(null, inBoundSessionId);
+        atConnection.metaData.isAuthenticated = true;
+        var syncVerbParams = HashMap<String, String>();
+        syncVerbParams.putIfAbsent(AT_FROM_COMMIT_SEQUENCE, () => '-1');
+        await syncProgressiveVerbHandler.processVerb(
+            response, syncVerbParams, atConnection);
+        List syncResponse = jsonDecode(response.data!);
+        expect(syncResponse.length, 1);
```

Comment:
Can we write this test in the following way:

1. We insert two keys in a way that only one of them fits into the buffer limit
     i.e size(key1) + size(key2) > buffer limit, however size(key1) < buffer limit
2.  Verify that the sync response contains only key1

Now increase the buffer limit in a way that accommodates both the keys
 i.e size(key1) + size(key2) < buffer limit
3.  Verify that the sync response contains both the keys

Reply:
This change is implemented.